scribus/scpainter.cpp

```diff
@@ -1955,8 +1962,18 @@ void ScPainter::drawText(QRectF area, QString text, bool filled, int align)
 	double ww = 0;
 	double hh = 0;
 	double r, g, b;
-	QString fontName = m_font.family() + " " + m_font.style();
-	cairo_select_font_face(m_cr, fontName.toLatin1(), CAIRO_FONT_SLANT_NORMAL, CAIRO_FONT_WEIGHT_NORMAL);
+
+	if (m_font.isNone())
```

Comment:
Why is that? The old code didn’t do any special handling if m_font was not set, I suggest to just return in this case. Even better add an assert here, so that anyone calling drawText without calling setFont d=first gets an assert (this whole API is smelly, but that is Scribus after all).